SemanticException: illegal attempt to dereference collection

## Description
<!-- Describe your issue -->
Error when opening the WordCreateController with the `autoFillText` parameter set.

```
[${web:attr.content_language}] 13:13:25.329 [qtp101478235-18] INFO  ai.elimu.web.content.word.WordCreateController - handleRequest
[${web:attr.content_language}] 13:13:25.329 [qtp101478235-18] INFO  ai.elimu.web.content.word.WordCreateController - autoSelectLetterSounds
[${web:attr.content_language}] 13:13:25.332 [qtp101478235-18] INFO  ai.elimu.web.servlet.CustomSimpleMappingExceptionResolver - logException
[${web:attr.content_language}] 13:13:25.332 [qtp101478235-18] ERROR ai.elimu.web.servlet.CustomSimpleMappingExceptionResolver - Request(GET http://eng.elimu.ai/content/word/create;jsessionid=node0gjh4n7c76tru5y6oronzjnmm908.node0?autoFillText=pollution)@506e50d8:)@407b9f96:)@3498e07c:)@5de5af3d:)@3498e07c:)@24b0b7e6:)@5de5af3d:)@49751c69:)@407b9f96:)@24b0b7e6:)@49751c69:)@3498e07c:)@49751c69
org.springframework.dao.InvalidDataAccessApiUsageException: org.hibernate.query.SemanticException: illegal attempt to dereference collection [ai.elimu.entity.content.LetterSound(lsc).letters] with element property reference [size]
        at org.springframework.orm.jpa.EntityManagerFactoryUtils.convertJpaAccessExceptionIfPossible(EntityManagerFactoryUtils.java:371) ~[spring-orm-6.0.9.jar:6.0.9]
        at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:234) ~[spring-orm-6.0.9.jar:6.0.9]
        at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:550) ~[spring-orm-6.0.9.jar:6.0.9]
        at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:61) ~[spring-tx-6.0.9.jar:6.0.9]
        at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:242) ~[spring-tx-6.0.9.jar:6.0.9]
        at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:152) ~[spring-tx-6.0.9.jar:6.0.9]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184) ~[spring-aop-6.0.9.jar:6.0.9]
        at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123) ~[spring-tx-6.0.9.jar:6.0.9]
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:391) ~[spring-tx-6.0.9.jar:6.0.9]
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119) ~[spring-tx-6.0.9.jar:6.0.9]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184) ~[spring-aop-6.0.9.jar:6.0.9]
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:223) ~[spring-aop-6.0.9.jar:6.0.9]
        at jdk.proxy4/jdk.proxy4.$Proxy221.readAllOrderedByLettersLength(Unknown Source) ~[?:?]
        at ai.elimu.web.content.word.WordCreateController.autoSelectLetterSounds(WordCreateController.java:185) ~[classes/:?]
        at ai.elimu.web.content.word.WordCreateController.handleRequest(WordCreateController.java:72) ~[classes/:?]
```

```
Caused by: java.lang.IllegalArgumentException: org.hibernate.query.SemanticException: illegal attempt to dereference collection [ai.elimu.entity.content.LetterSound(lsc).letters] with element property reference [size]
        at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:138) ~[hibernate-core-6.1.7.Final.jar:6.1.7.Final]
        at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:175) ~[hibernate-core-6.1.7.Final.jar:6.1.7.Final]
        at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:182) ~[hibernate-core-6.1.7.Final.jar:6.1.7.Final]
        at org.hibernate.internal.AbstractSharedSessionContract.createQuery(AbstractSharedSessionContract.java:760) ~[hibernate-core-6.1.7.Final.jar:6.1.7.Final]
        at org.hibernate.internal.AbstractSharedSessionContract.createQuery(AbstractSharedSessionContract.java:662) ~[hibernate-core-6.1.7.Final.jar:6.1.7.Final]
        at org.hibernate.internal.AbstractSharedSessionContract.createQuery(AbstractSharedSessionContract.java:126) ~[hibernate-core-6.1.7.Final.jar:6.1.7.Final]
        at jdk.internal.reflect.GeneratedMethodAccessor87.invoke(Unknown Source) ~[?:?]
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.base/java.lang.reflect.Method.invoke(Method.java:569) ~[?:?]
        at org.springframework.orm.jpa.SharedEntityManagerCreator$SharedEntityManagerInvocationHandler.invoke(SharedEntityManagerCreator.java:311) ~[spring-orm-6.0.9.jar:6.0.9]
        at jdk.proxy4/jdk.proxy4.$Proxy214.createQuery(Unknown Source) ~[?:?]
        at ai.elimu.dao.jpa.LetterSoundDaoJpa.readAllOrderedByLettersLength(LetterSoundDaoJpa.java:50) ~[classes/:?]
```

## Why?
<!-- Explain _why_ this issue is needed -->
Solving this bug is needed so hat 

---

> elimu.ai's mission is to build innovative learning software that empowers out-of-school children to teach themselves basic reading📖, writing✍🏽 and math🔢 **within 6 months**.

--- a/pom-dependency-tree.txt
+++ b/pom-dependency-tree.txt
@@ -1,4 +1,4 @@
-ai.elimu:webapp:war:2.6.85-SNAPSHOT
+ai.elimu:webapp:war:2.6.86-SNAPSHOT
 +- ai.elimu:model:jar:model-2.0.114:compile
 |  \- com.google.code.gson:gson:jar:2.13.1:compile
 |     \- com.google.errorprone:error_prone_annotations:jar:2.38.0:compile

--- a/src/main/java/ai/elimu/dao/jpa/LetterSoundDaoJpa.java
+++ b/src/main/java/ai/elimu/dao/jpa/LetterSoundDaoJpa.java
@@ -48,9 +48,9 @@ public class LetterSoundDaoJpa extends GenericDaoJpa<LetterSound> implements Let
     @Override
     public List<LetterSound> readAllOrderedByLettersLength() throws DataAccessException {
         return em.createQuery(
-            "SELECT lsc " +
-            "FROM LetterSound lsc " +
-            "ORDER BY lsc.letters.size DESC, lsc.usageCount DESC")
+            "SELECT ls " +
+            "FROM LetterSound ls " +
+            "ORDER BY SIZE(ls.letters) DESC, ls.usageCount DESC")
             .getResultList();
     }
 }

--- a/src/main/webapp/WEB-INF/jsp/content/storybook/edit.jsp
+++ b/src/main/webapp/WEB-INF/jsp/content/storybook/edit.jsp
@@ -310,7 +310,7 @@
                         <c:choose>
                             <c:when test="${empty wordMap[wordTextLowerCase]}">
                                 <c:out value="${wordText}" /><br />
-                                <a href="<spring:url value='/content/word/create?autoFillText=${wordTextLowerCase}' />" target="_blank">Add word <i class="material-icons">launch</i></a>
+                                <a href="<spring:url value='/content/word/create?autoFillText=${wordTextLowerCase}' />" class="autoFillWordLink" target="_blank">Add word <i class="material-icons">launch</i></a>
                             </c:when>
                             <c:otherwise>
                                 <c:set var="word" value="${wordMap[wordTextLowerCase]}" />

--- a/src/test/java/selenium/content/storybook/StoryBookEditPage.java
+++ b/src/test/java/selenium/content/storybook/StoryBookEditPage.java
@@ -1,7 +1,11 @@
 package selenium.content.storybook;
 
+import java.util.List;
+import java.util.Set;
+
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
 
 public class StoryBookEditPage {
     
@@ -11,5 +15,22 @@ public class StoryBookEditPage {
         this.driver = driver;
 
         driver.findElement(By.id("storyBookEditPage"));
+    }
+
+    public void pressRandomAutoFillWordLink() {
+        List<WebElement> links = driver.findElements(By.className("autoFillWordLink"));
+        int randomIndex = (int) (Math.random() * links.size());
+        WebElement randomLink = links.get(randomIndex);
+        randomLink.click();
+
+        // Switch to the new window
+        String mainWindowHandle = driver.getWindowHandle();
+        Set<String> windowHandles = driver.getWindowHandles();
+        for (String handle : windowHandles) {
+            if (!handle.equals(mainWindowHandle)) {
+                driver.switchTo().window(handle);
+                break;
+            }
+        }
     }
 }

--- a/src/test/java/selenium/content/word/WordTest.java
+++ b/src/test/java/selenium/content/word/WordTest.java
@@ -9,6 +9,8 @@ import org.openqa.selenium.chrome.ChromeOptions;
 
 import lombok.extern.slf4j.Slf4j;
 import selenium.content.MainContentPage;
+import selenium.content.storybook.StoryBookEditPage;
+import selenium.content.storybook.StoryBookListPage;
 import selenium.util.DomainHelper;
 
 @Slf4j
@@ -68,5 +70,24 @@ public class WordTest {
         wordListPage.pressCreateButton();
 
         WordCreatePage wordCreatePage = new WordCreatePage(driver);
+    }
+
+    @Test
+    public void testWordAutoFill() {
+        log.info("testWordAutoFill");
+
+        MainContentPage mainContentPage = new MainContentPage(driver);
+        mainContentPage.pressStoryBookListLink();
+
+        StoryBookListPage storyBookListPage = new StoryBookListPage(driver);
+        storyBookListPage.pressRandomStoryBook();
+        log.info("driver.getCurrentUrl(): " + driver.getCurrentUrl());
+
+        StoryBookEditPage storyBookEditPage = new StoryBookEditPage(driver);
+        storyBookEditPage.pressRandomAutoFillWordLink();
+        log.info("driver.getCurrentUrl(): " + driver.getCurrentUrl());
+
+        WordCreatePage wordCreatePage = new WordCreatePage(driver);
+        // TODO
     }
 }

--- a/src/test/java/selenium/content/word/WordTest.java
+++ b/src/test/java/selenium/content/word/WordTest.java
@@ -88,6 +88,6 @@ public class WordTest {
         log.info("driver.getCurrentUrl(): " + driver.getCurrentUrl());
 
         WordCreatePage wordCreatePage = new WordCreatePage(driver);
-        // TODO
+        // TODO: assert auto-filled word text
     }
 }

